cats/check_clean_arguments.py

```diff
@@ -20,41 +21,50 @@ def validate_jobinfo(jobinfo: str):
         "cpus",
         "gpus",
     )
-    info = {
-        k: v
-        for k, v in [match.groups() for match in re.finditer(r"(\w+)=(\w+)", jobinfo)]
-    }
-    missing_keys = set(expected_info_keys) - set(info.keys())
-    if missing_keys:
-        print(f"ERROR: Missing job info keys: {missing_keys}")
+    info = dict([match.groups() for match in re.finditer(r"(\w+)=(\w+)", jobinfo)])
+
+    # Check if some information is missing
+    if missing_keys := set(expected_info_keys) - set(info.keys()):
+        sys.stderr.write(f"ERROR: Missing job info keys: {missing_keys}")
         return {}
-    expected_partition_values = ("CPU_partition", "GPU_partition")
+
+    # Validate partition value
+    expected_partition_values = config['partitions'].keys()
     if info["partition"] not in expected_partition_values:
-        msg = (
-            "ERROR: job info key 'partition' should be "
-            f"one of {expected_partition_values}. Typo?"
-        )
-        print(msg)
+        sys.stderr.write("ERROR: job info key 'partition' should be one of {expected_partition_values}. Typo?\n")
         return {}
-    for key in [k for k in info.keys() if k != "partition"]:
+
+    # check that `cpus`, `gpus` and `memory` are numeric and convert to int
+    for key in [k for k in info if k != "partition"]:
         try:
             info[key] = int(info[key])
         except ValueError:
-            print(f"ERROR: job info key {key} should be numeric")
+            sys.stderr.write(f"ERROR: job info key {key} should be numeric\n")
             return {}
+
     return info
 
 def validate_duration(duration):
-    # make sure size is not None
-    if duration is None:
-        raise ValueError("Windowed method requires timespan to be provided")
-    # make sure size is can be converted to integer
+    # make sure it can be converted to integer
     try:
         duration_int = int(duration)
     except ValueError:
-        raise ValueError("Windowed method requires timespan to be an integer or float")
-    # make sure size is positive
+        raise ValueError("--duration needs to be an integer or float (number of minutes)")
+    # make sure it's positive
     if duration_int <= 0:
-        raise ValueError("Windowed method requires timespan to be positive")
+        raise ValueError("--duration needs to be positive (number of minutes)")
 
     return duration_int
+
+def validate_location(location, choice_CI_API):
+    if choice_CI_API == 'carbonintensity.org.uk':
```

Comment:
Ok for now, but this indicates we need a validation function alongside url and response parsing when defining an API interface

Reply:
Indeed good catch, I've added a TODO so we don't forget